tests/test_rank.py

```diff
@@ -403,42 +392,139 @@ def test_get_knn_distance_ood_scores():
-            features=X_test_with_ood, nbrs=nbrs, k=10  # this should throw an assertion error
+    avg_knn_distances_default_model = rank.get_outlier_scores(
+        features=X_with_ood,
+        k=instantiated_k,  # this should use default estimator (same as above) and k = instantiated_k
+    )
+
+    avg_knn_distances_default_k = rank.get_outlier_scores(
+        features=X_with_ood  # default k should be set to 10 == instantiated_k
+    )
+
+    avg_knn_distances = rank.get_outlier_scores(
+        features=None,
+        knn=knn,
+        k=25,  # this should throw user warn, k should be set to instantiated_k
+    )
+
+    # Score sums should be equal because the three estimators used have identical params and fit
+    assert avg_knn_distances.sum() == avg_knn_distances_default_model.sum()
+    assert avg_knn_distances_default_k.sum() == avg_knn_distances.sum()
+
+    avg_knn_distances_large_k = rank.get_outlier_scores(
+        features=X_with_ood,
+        k=25,  # this should use default estimator and k = 25
+    )
+
+    avg_knn_distances_tiny_k = rank.get_outlier_scores(
+        features=None,
+        knn=knn,
+        k=1,  # this should use knn estimator and k = 1
+    )
+
+    avg_knn_distances_tiny_k_default = rank.get_outlier_scores(
+        features=X_with_ood,
+        k=1,  # this should use default estimator and k = 1
+    )
+
+    # Score sums should be different because k = user param for estimators and k != 10.
+    assert avg_knn_distances_tiny_k.sum() != avg_knn_distances.sum()
+    assert avg_knn_distances_large_k.sum() != avg_knn_distances.sum()
+    assert avg_knn_distances_tiny_k_default.sum() != avg_knn_distances_default_model.sum()
+
+    # Test that when knn is None ValueError raised if passed in k > len(features)
+    try:
+        rank.get_outlier_scores(
+            features=X_with_ood,
+            knn=None,
+            k=len(X_with_ood) + 1,  # this should throw ValueError, k ! > len(features)
+        )
+    except Exception as e:
+        assert "nearest neighbors" in str(e)
+        with pytest.raises(ValueError) as e:
+            rank.get_outlier_scores(
+                features=X_with_ood,
+                knn=None,
+                k=len(X_with_ood) + 1,  # this should throw ValueError, k ! > len(features)
+            )
+
+
+def test_not_enough_info_get_outlier_scores():
+    # Testing calling function with not enough information to calculate outlier scores
+    try:
+        rank.get_outlier_scores(
+            features=None,
+            knn=None,
+            k=15,  # this should throw TypeError because knn=None and features=None
+        )
+    except Exception as e:
+        assert "Both knn and features arguments" in str(e)
+        with pytest.raises(ValueError) as e:
+            rank.get_outlier_scores(
+                features=None,
+                knn=None,
+                k=15,  # this should throw TypeError because knn=None and features=None
+            )
```

Comment:
This seems basically the same logic as the below block except for the `k=15` argument. Is it really testing something different and thus worth keeping?

Reply:
Testing different k does not affect exception throw but there is no real reason for this (pretty sure it doesnt/wouldnt)

Reply:
Ya let’s get rid of this block then. It’s a little confusing to have two different blocks that are testing the same thing